raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -207,13 +229,37 @@ public void updateElectionStartMs(long currentTimeMs) {
         electionStartMs = OptionalLong.of(currentTimeMs);
     }
 
+    public void updateNumVoters(int numVoters) {
+        this.numVoters = numVoters;
+    }
+
+    public void updateNumObservers(int numObservers) {
+        this.numObservers = numObservers;
+    }
+
+    public void updateUncommittedVoterChange(boolean uncommittedVoterChange) {
+        this.uncommittedVoterChange = uncommittedVoterChange;
+    }
+
     public void maybeUpdateElectionLatency(long currentTimeMs) {
         if (electionStartMs.isPresent()) {
             electionTimeSensor.record(currentTimeMs - electionStartMs.getAsLong(), currentTimeMs);
             electionStartMs = OptionalLong.empty();
         }
     }
 
+    public void addLeaderMetrics() {
+        metrics.addMetric(numObserversMetricName, (Gauge<Integer>) (config, now) -> numObservers);
+        metrics.addMetric(uncommittedVoterChangeMetricName, (Gauge<Boolean>) (config, now) -> uncommittedVoterChange);
```

Comment:
This exposes the metrics as false/true. The KIP mentions that the metric is meant to expose 0/1.